### ISSUE TITLE
New check: Check if dynamic exceptions are ignored

### ISSUE THREAD

**Issue by ConjuringCoffee** [larshp/abapOpenChecks#1118]
If a method raises an exception inheriting from CX_DYNAMIC_CHECK, then the caller is not forced to catch or raise the exception themselves. For static exceptions, a syntax warning is shown if the caller ignores the exception - but there's absolutely no indication for dynamic exceptions.

I'd like to have a check that can show all instances in which a dynamic exception was ignored. I wouldn't include this in any quality gate, but I'd use it just to ensure that I didn't ignore an exception on accident. 

I think we'd have to:

- Find all method calls 
- For each method call, get its signature
- For each method signature, get all exception classes
- For each exception class, go through the inheritance tree to find out if it is a dynamic exception.
- (Stop processing any method call without a dynamic exception.)
- For each exception, check if the current method is raising the exception (or one of the parent classes)
- If not, check the current method for a CATCH with the exception (or one of the parent classes)

Unfortunately, that sounds really complicated and not really feasible to me. The first steps already seem really complicated (detecting all method calls and getting their signature). 

Do you have any better idea for how this kind of check could be achieved? Maybe I'm making this seem more complicated than it actually is. 

Could it be possible to somehow leverage the already existing logic for the syntax warning for ignored static exceptions?

**Comment by larshp** [comment]
Hi, good idea, its kind of a mess to build in code inspector, plus have working on multiple versions.

So, I've added a setting to https://rules.abaplint.org/uncaught_exception/ to have it give errors for dynamic or no check exceptions.

However, personally, I'd like to keep track and overview of the dynamic exceptions during development, so I've added the following code lens. Note: it currently always shows the dynamic exception, even if they are caught or propagated, I'll probably change that in the future, first step is finding the basic information correctly,
![image](https://github.com/larshp/abapOpenChecks/assets/5888506/e3952ac3-9007-4c7f-9091-daed8c33dbab)
